### PR TITLE
CAT-1886 Add sensor variable nfc_tag_id

### PR DESCRIPTION
**Pull request by JoachimLesser**
stores last read NFC Tag in new sensor variable nfc_tag_id.

### CODE REVIEW

File: catroid/src/org/catrobat/catroid/nfc/NfcHandler.java [hunk 1]
```diff
@@ -59,44 +66,23 @@ public static String getUid(Intent intent) {
...
 	}
 
-	public static String byteArrayToHex(byte[] a) {
-		if (a == null) {
-			return null;
+	public static int byteArrayToInt(byte[] b) {
+		int value = 0;
+		for (int i = 0; i < 4; i++) {
+			int shift = (4 - 1 - i) * 8;
+			value += (b[i] & 0x000000FF) << shift;
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/nfc/NfcHandler.java]: Choose one of the following lines instead :wink:
```java
Ints.fromByteArray(b);
ByteBuffer.wrap(b).getInt();
(b[0] & 0xFF) << 24 | (b[1] & 0xFF) << 16 | (b[2] & 0xFF) << 8  | (b[3] & 0xFF);
```

File: catroid/src/org/catrobat/catroid/nfc/NfcHandler.java [hunk 2]
```diff
@@ -59,44 +66,23 @@ public static String getUid(Intent intent) {
...
 		}
-		StringBuilder sb = new StringBuilder();
-		for (byte b : a) {
-			sb.append(String.format("%02x", b & 0xff));
-		}
-		return sb.toString();
+		return Math.abs(value);
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/nfc/NfcHandler.java]: Remove `Math.abs` if possible.

File: catroid/src/org/catrobat/catroid/common/NfcTagData.java [hunk 3]
```diff
@@ -63,6 +63,7 @@ public String getNfcTagUid() {
 		return uid;
 	}
 
+
```
Comment (thmq) [inline catroid/src/org/catrobat/catroid/common/NfcTagData.java]: this line is unnecessary and should be removed (one new line suffices).

**Comment by thmq** [comment]
there are merge conflicts here, please rebase and reopen when done.

**Pull request closed without merging**